### PR TITLE
feat(content,port): Arthropod Overhaul Part 1, Dragonfly

### PR DESCRIPTION
**Pull request by Zlorthishen**
## Purpose of change
ports over dda changes:
https://github.com/CleverRaven/Cataclysm-DDA/pull/40733 by https://github.com/ymber
https://github.com/CleverRaven/Cataclysm-DDA/pull/44779 by https://github.com/Venera3

## Describe the solution
 - adds new insect organs
 - adds new text snippets 
 - changes harvests to allow for these new snippets and items
 - add new variations of dragonfly species
 - add new `INSECT_FLYING` faction

## Testing
image

image

image

### CODE REVIEW

**Comment by chaosvolt** [comment]
Is it supposed to print so many snippets at once?

File: data/json/harvest.json [hunk 1]
```diff
@@ -443,75 +443,125 @@
...
+      { "drop": "endochitin", "type": "bone", "mass_ratio": 0.1 },
+      { "drop": "acidchitin_piece", "type": "bone", "mass_ratio": 0.1 }
     ]
   },
   {
     "id": "arachnid_bee",
-    "//": "todo: add stinger here and remove drops from death",
     "type": "harvest",
-    "message": "You carefully crack open its exoskeleton to get at the flesh beneath",
+    "message": "What appeared to be insect hairs on the chitin of this creature look more like tiny feathers as you pare them back.  Inside is a bundle of bubble-like tissue sacs that appear to be floating, which doesn't fit with what you know about real bees.",
```
Comment (chaosvolt) [inline data/json/harvest.json:509]: Not a snippet thingy?

File: data/json/harvest.json [hunk 2]
```diff
@@ -443,75 +443,125 @@
...
+      { "drop": "mutant_bug_lungs", "type": "flesh", "mass_ratio": 0.0035 },
+      { "drop": "mutant_bug_organs", "type": "offal", "mass_ratio": 0.015 },
+      { "drop": "wasp_sting", "base_num": [ 0, 1 ], "type": "bone" },
+      { "drop": "chitin_piece", "type": "bone", "mass_ratio": 0.1 }
+    ]
+  },
+  {
+    "id": "arachnid_dragonfly_mega",
+    "type": "harvest",
+    "message": "As you hack away at the chitinous abomination you can't help but follow the tunnel the (male?) dragonfly's tail bore into the flesh of its mate, ending up at a formless amalgamation of eyes, jaws, eggs, and ganglia deep within.  Love truly knows no bounds.  <arachnid_flying_harvest>",
```
Comment (chaosvolt) [inline data/json/harvest.json:539]:
```suggestion
    "message": "<arachnid_flying_harvest>",
```

**Review by chaosvolt — DISMISSED** [review]
Think that's everything aside from hopefully creating snippets for bees and wasps in a future PR, per discussion on discord.

File: data/json/monsters/insect_spider.json [hunk 3]
```diff
@@ -400,37 +399,116 @@
...
+    "id": "mon_dragonfly_mega",
+    "type": "MONSTER",
+    "name": { "str": "mega dragonfly", "str_pl": "mega dragonflies" },
+    "description": "A single, enormous insect with a swollen abdomen and a distended ovipositor, periodically dropping a handful of glistening eggs.",
+    "volume": "600 L",
+    "weight": "800 kg",
+    "default_faction": "dragonfly",
+    "bodytype": "insect",
+    "species": [ "INSECT_FLYING" ],
+    "hp": 1100,
```
Comment (chaosvolt) [inline data/json/monsters/insect_spider.json:487]: I just now however noticed that is more than twice the HP of hulks, dafuq? Especially when the PR already nerfs regular giant dragonflies down from 70 to 50 HP.
```suggestion
    "hp": 110,
```
I assume this was an accident and it's off by a factor of 10.

**Review by chaosvolt — APPROVED** [review]
stonks

**Comment by scarf005** [comment]
awesome. next time maybe could we add [Co-authored-by](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) to individual commits? this helps properly attributing original authors. for example:

```
Co-authored-by: ymber <mtaylor1252dev@gmail.com>
Co-authored-by: Venera3 <72006894+Venera3@users.noreply.github.com>
```

**Pull request merged**